### PR TITLE
[RFR] Fixing client-side petition groupings

### PR DESCRIPTION
**Pull request by eschaefer**
This just un-breaks the mess I made earlier. Still not great, but I don't want to waste more time on it.

### CODE REVIEW

File: src/containers/Home.js [hunk 1]
```diff
@@ -16,8 +16,7 @@ const HomeContainer = React.createClass({
   componentWillMount () {
     // If there are no petitions, or if the user arrived on the page by clicking
     // a client-side router link, then we fetch petitions client-side.
-    if (!hasPetitionGroups(this.props.petitions) || isClientSideRouting(this.props.location)) {
-      this.props.clearPetitions();
+    if (!hasPetitionGroups(this.props) || isClientSideRouting(this.props.location)) {
```
Comment (mattberridge) [inline src/containers/Home.js]: think we can bin the `isClientSideRouting(this.props.location)` from here no?

File: src/helpers/getPetitionsQueryParams.js [hunk 2]
```diff
@@ -4,5 +4,5 @@ export default (params = {}, query = {}) => ({
   cityName: params.cityName || '',
   limit: parseInt(query.limit || 12),
   state: query.state || 'current',
-  sort: query.sort || ''
+  sort: query['sort'] || ''
```
Comment (mattberridge) [inline src/helpers/getPetitionsQueryParams.js]: why the change here, out of interest?

File: src/actions/PetitionsActions.js [hunk 3]
```diff
@@ -60,14 +67,18 @@ export function requestPetitions () {
...
 
-export function receivePetitions (petitions, params, qs, group) {
-  const groupedPetitions = group
-    ? { [group]: petitions }
-    : petitions;
+export function receiveGroupedPetitions (petitions, params, group) {
+  return {
+    type: RECEIVE_GROUPED_PETITIONS,
+    params,
+    groupedPetitions: { [group]: petitions }
```
Comment (mattberridge) [inline src/actions/PetitionsActions.js]: I would still maybe do this in the reducer if possible as it's assigning state

File: src/selectors/petitionsGroups.js [hunk 4]
```diff
@@ -1,12 +1,11 @@
...
 import getPetitions from 'selectors/petitions';
 
-export default (petitions = {}, groups = []) => {
+export default (petitions, groups) => {
   const petitionGroups = {};
 
-  forEach(groups, (group) =>
-    set(petitionGroups, group, getPetitions(petitions[group].data || []))
+  groups.forEach((group) =>
+    set(petitionGroups, group.group, getPetitions(petitions[group.group] && petitions[group.group].data || []))
```
Comment (mattberridge) [inline src/selectors/petitionsGroups.js]: Quite irrelevant but not a fan of the `group.group` naming. But don't change it if there's nothing better :)

**Comment by mattberridge** [comment]
Nicely clean and flexible solution in the end here. A few comments you might examine but overall pretty well good to go

👍

File: src/actions/PetitionsActions.js [hunk 5]
```diff
@@ -29,19 +29,26 @@ export function fetchPetitions ({ location, params }, group) {
...
 }
 
 export function fetchGroupedPetitions (props = {}, groups = []) {
-  const sortQuery = getHomePagePetitionsQuery();
-  const location = Object.assign({}, props.location, sortQuery);
-
   return (dispatch) => {
-    const groupDispatches = groups.map(group =>
-      dispatch(fetchPetitions({ location }, group)));
+    dispatch(requestPetitions());
```
Comment (mattberridge) [inline src/actions/PetitionsActions.js]: Not sure this logic for `isLoading` is re-useable here actually.
We set it to `petitions: { isLoading: true }` here and then after the first group finishes it's set to `false`?

Do you think we need;

```
latest: {
  isLoading
},
trending: {
  isLoading
}
```

so we could display a loading spinner for each group individually?

Not critical, I guess

File: src/containers/Home.js (hunk 1 above)
Comment (eschaefer) [inline src/containers/Home.js]: yep

File: src/helpers/getPetitionsQueryParams.js (hunk 2 above)
Comment (eschaefer) [inline src/helpers/getPetitionsQueryParams.js]: My syntax highlighting was showing that `query.sort` was corresponding to the native `sort()` method, so I was OCD and changed it :P

File: src/actions/PetitionsActions.js (hunk 5 above)
Comment (eschaefer) [inline src/actions/PetitionsActions.js]: Yep, you're right. Each group needs its own `isLoading` property. And  the `<Loading />` component needs to be wrapped around the homepage petitions...
